src/SecureStore.BeyondTrust/BeyondTrustDynamicSystemSecureStore.cs

```diff
@@ -0,0 +1,146 @@
+﻿using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Threading.Tasks;
+using UiPath.Orchestrator.Extensibility.Configuration;
+using UiPath.Orchestrator.Extensibility.SecureStores;
+
+namespace UiPath.Orchestrator.Extensions.SecureStores.BeyondTrust
+{
+    public class BeyondTrustDynamicSystemSecureStore : ISecureStore
+    {
+        public void Initialize(Dictionary<string, string> hostSettings)
+        {
+            // No host level settings
+        }
+
+        public Task ValidateContextAsync(string context)
+        {
+            BeyondTrustVaultClientFactory.GetClient(context).TestConnection();
+            return Task.CompletedTask;
+        }
+
+        public IEnumerable<ConfigurationEntry> GetConfiguration() => new List<ConfigurationEntry>
+            {
+                new ConfigurationValue(ConfigurationValueType.String)
+                {
+                    Key = ConfigurationConstants.Hostname,
+                    DisplayName = "BeyondTrust Host URL",
+                    IsMandatory = true,
+                },
+                new ConfigurationValue(ConfigurationValueType.String)
+                {
+                    Key = ConfigurationConstants.AuthKey,
+                    DisplayName = "API Authentication Key",
+                    IsMandatory = true,
+                },
+                new ConfigurationValue(ConfigurationValueType.String)
+                {
+                    Key = ConfigurationConstants.RunAs,
+                    DisplayName = "API Run As",
+                    IsMandatory = true,
+                },
+                new ConfigurationValue(ConfigurationValueType.Boolean)
+                {
+                    Key = ConfigurationConstants.SSLEnabled,
+                    DisplayName = "Use SSL certificate",
+                    IsMandatory = true,
+                },
+                new ConfigurationValue(ConfigurationValueType.String)
+                {
+                    Key = ConfigurationConstants.SystemAccountDelimiter,
+                    DisplayName = "System-Account Delimiter",
+                    IsMandatory = true,
+                    DefaultValue = "/",
+                },
+                new ConfigurationValue(ConfigurationValueType.Choice)
+                {
+                    Key = ConfigurationConstants.ManagedAccountType,
+                    DisplayName = "Managed Account Type",
+                    IsMandatory = true,
+                    PossibleValues = new List<string> { ConfigurationConstants.SystemAccountType, ConfigurationConstants.DomainAccountType }
+                },
+            };
+
+        public SecureStoreInfo GetStoreInfo()
+        {
+            return new SecureStoreInfo { Identifier = "BeyondTrust-DynamicSystem-ReadOnly", IsReadOnly = true };
```

Comment:
The customers have requested a read/write version as well. The development for that has been paused to prioritize the read-only version, but it will get picked back up as soon as this gets published.